File: tmp/leetcode/121_best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        l, r = 0, 0
        ans = 0
        while r < len(prices):
            if prices[l] >= prices[r]:
                l = r
            ans = max(ans, prices[r] - prices[l])
            r += 1
            
        return ans

File: tmp/leetcode/121_best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l=0, r=0;
        int res=0;
        
        while (r < prices.length) {
            if (l == r) {
                r++;
                continue;
            }
            
            if (prices[l] > prices[r]) {
                l = r++;
                continue;
            }
            
            res = Math.max(res, prices[r] - prices[l]);
            r++;
        }
        
        return res;
    }
}
